ra-optimisation/search-tree-rao/src/test/java/com/farao_community/farao/search_tree_rao/state_tree/StateTreeTest.java

```diff
@@ -0,0 +1,370 @@
+
+        ContingencyScenario contingencyScenario = stateTree.getContingencyScenarios().iterator().next();
+        assertEquals(crac.getContingency("contingency-1"), contingencyScenario.getContingency());
+        assertTrue(contingencyScenario.getAutomatonState().isEmpty());
+        assertEquals(crac.getState("contingency-1", Instant.CURATIVE), contingencyScenario.getCurativeState());
+    }
+
+    @Test
+    public void testCreatePerimetersWithTwoRemedialActions() {
+        setUpCustomCrac();
+        crac.newPstRangeAction()
+            .withId("pst-ra-1")
+            .withNetworkElement("pst1")
+            .withInitialTap(1)
+            .withTapToAngleConversionMap(Map.of(1, 1., 2, 2.))
+            .newOnStateUsageRule().withContingency("contingency-1").withInstant(Instant.CURATIVE).withUsageMethod(UsageMethod.AVAILABLE).add()
+            .add();
+        crac.newPstRangeAction()
+            .withId("pst-ra-2")
+            .withNetworkElement("pst2")
+            .withInitialTap(1)
+            .withTapToAngleConversionMap(Map.of(1, 1., 2, 2.))
+            .newOnStateUsageRule().withContingency("contingency-2").withInstant(Instant.CURATIVE).withUsageMethod(UsageMethod.AVAILABLE).add()
+            .add();
+        stateTree = new StateTree(crac);
+        assertEquals(5, stateTree.getBasecaseScenario().getAllStates().size());
+        assertEquals(2, stateTree.getContingencyScenarios().size());
+    }
+
+    @Test
+    public void testCreatePerimetersWithTwoRemedialActionsOnSameContingency() {
+        setUpCustomCrac();
+        crac.newPstRangeAction()
+            .withId("pst-ra-1")
+            .withNetworkElement("pst1")
+            .withInitialTap(1)
+            .withTapToAngleConversionMap(Map.of(1, 1., 2, 2.))
+            .newOnStateUsageRule().withContingency("contingency-2").withInstant(Instant.CURATIVE).withUsageMethod(UsageMethod.AVAILABLE).add()
+            .add();
+        crac.newPstRangeAction()
+            .withId("pst-ra-2")
+            .withNetworkElement("pst2")
+            .withInitialTap(1)
+            .withTapToAngleConversionMap(Map.of(1, 1., 2, 2.))
+            .newOnStateUsageRule().withContingency("contingency-2").withInstant(Instant.CURATIVE).withUsageMethod(UsageMethod.AVAILABLE).add()
+            .add();
+        stateTree = new StateTree(crac);
+        assertEquals(6, stateTree.getBasecaseScenario().getAllStates().size());
+        assertEquals(1, stateTree.getContingencyScenarios().size());
+    }
+
+    @Test
+    public void testErrorOnOutageRa() {
+        setUpCustomCrac();
+        Crac mockCrac = Mockito.spy(crac);
+        State outageState = crac.getState("contingency-1", Instant.OUTAGE);
+        Mockito.when(mockCrac.getNetworkActions(Mockito.eq(outageState), Mockito.any()))
+            .thenReturn(Set.of(Mockito.mock(NetworkAction.class)));
+        assertThrows(FaraoException.class, () -> new StateTree(mockCrac));
+    }
```

Comment:
I don't understand what is tested here ? :thinking:

Reply:
that an outage state with a remedial action throws an exception in the statetree

Reply:
(even though it's not permitted in CracImpl's usage rules adders, we'd throw an exception in the RAO just in case)

Reply:
I updated it to make it a little more clear

Reply:
ok, thanks for the explanation :)